src/app/Dashboard/DraggableRef.tsx

```diff
@@ -35,125 +35,327 @@
+      if (hoveringDroppable.current && hoveringIndex.current !== null) {
+        setIsDragging(false);
+        setRefStyle({
+          ...initStyle,
+          transition: transition,
+          transform: '',
+        });
+        dispatch(dashboardCardConfigReorderCardIntent(dashboardId, hoveringIndex.current, swap.current));
       } else {
-        console.error('cardRef.current is undefined');
+        setRefStyle({
+          ...refStyle,
+          transition: transition,
+          transform: '',
+        });
       }
     },
-    [dispatch, cardRef, cardConfigs, nearEdgeMultiplier, dashboardId, cardSizes]
+    [dispatch, setIsDragging, setRefStyle, refStyle, dashboardId]
   );
 
-  const callbackMouseUp = React.useCallback(() => {
-    if (!isResizing.current) {
-      return;
-    }
-    isResizing.current = false;
-    document.body.style.removeProperty('cursor');
-    document.removeEventListener('mousemove', callbackMouseMove);
-    document.removeEventListener('mouseup', callbackMouseUp);
-  }, [callbackMouseMove]);
-
-  const handleOnMouseDown = React.useCallback(
-    (e: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
-      e.stopPropagation();
-      e.preventDefault();
-      document.body.style.setProperty('cursor', 'col-resize');
-      document.addEventListener('mousemove', callbackMouseMove);
-      document.addEventListener('mouseup', callbackMouseUp);
-      isResizing.current = true;
-      minWidth.current = undefined;
-      maxWidth.current = undefined;
+  const onMouseMoveWhileDragging = React.useCallback(
+    (ev: MouseEvent, droppableItems: DroppableItem[]) => {
+      hoveringDroppable.current = null;
+      hoveringIndex.current = null;
+      const currDragged = wrapperRef.current;
+      if (currDragged) {
+        isMouseDown.current = true;
+        const dragOrder = getOrder(currDragged);
+        droppableItems.forEach((di, idx) => {
+          // mouse is hovering on a card
+          if (!di.isDraggingHost && overlaps(ev, di.rect)) {
+            setDroppableItem(di, transition, `translate(0, ${overlapTranslateY}px`, false);
+            hoveringDroppable.current = di.node;
+            hoveringIndex.current = idx;
+            swap.current = true;
+            // reset non overlapping items
+            droppableItems.forEach((_item, _idx) => {
+              if (_idx == idx || _item.isDraggingHost) return;
+              resetDroppableItem(_item);
+            });
+          } else {
+            // mouse is hovering between two cards
+            const nextItem = droppableItems[(idx + 1) % droppableItems.length];
+            const [betweenTwoRects, draggedPosition] = inBetween(ev, di.rect, nextItem.rect);
+            if (betweenTwoRects && droppableItems.length > 1) {
+              const hover = (idx + 1) % droppableItems.length;
+              // if the dragged card has a greater order than the hovered card
+              if (dragOrder > hover) {
```

Comment:
Probably revise `move` function to remove duplicate case. And remove nested loops